boa/explorer.py

```diff
@@ -28,17 +39,33 @@ def _fetch_etherscan(
         res = SESSION.get(uri, params=params)
         res.raise_for_status()
         data = res.json()
-        if not (data.get("status") == "0" and "rate limit" in data.get("result", "")):
+        if not _is_rate_limited(data):
             break
         backoff_factor = 1.1**i  # 1.1**10 ~= 2.59
         time.sleep(backoff_factor * backoff_ms / 1000)
 
-    if int(data["status"]) != 1:
+    if not _is_success_response(data):
         raise ValueError(f"Failed to retrieve data from API: {data}")
 
     return data
 
 
+def _is_success_response(data: dict) -> bool:
+    return int(data["status"]) == 1
```

Comment:
honestly i think this should be
```suggestion
    return data.get("status") == "1"
```
the previous code could blow up (i.e. raise an unexpected/unrecoverable error) if data.status is not an int or not present

Reply:
I think it would be worth to have a failure if we receive unexpected stuff, other than failing silently

Reply:
hm i guess that's a good point. i'm neutral on this then